frontend/src/routes/Search.tsx

```diff
@@ -82,12 +113,22 @@ const query = graphql`
 
 type Props = {
     q: string;
+    f: string;
+    start: string;
+    end: string;
     outcome: SearchQuery$data["search"];
 };
 
-const SearchPage: React.FC<Props> = ({ q, outcome }) => {
+const SearchPage: React.FC<Props> = ({ q, f, start, end, outcome }) => {
     const { t } = useTranslation();
     const router = useRouter();
+    const defaultFilter = f as Filter;
+
+    const [filter, setFilter] = useState<Filter>(defaultFilter);
+    const filterType: Filter[] = ["all", "videos", "pages"];
+
+    const [startDate, setStartDate] = useState(start);
+    const [endDate, setEndDate] = useState(end);
```

Comment:
Regarding those `useState` usages: I don't think they are necessary? All changes to filters result in a router `goto` with the new values in the URL. This is the same for `q`